### ISSUE TITLE
Is libdivsufsort expected to be cloned separately?

### ISSUE THREAD

**Issue by rsharris** [simongog/sdsl-lite#318]
Howdy,

With gcc 4.9 now installed, I threw away my old clone (see issue #315), cloned anew, and attempted to build. This now fails because libdivsufsort is an empty directory.

I've appended to output of install, below.

I'm not sure if this is a problem with the clone (I tried several times, libdivsufsort was empty each time) or if the user is expected to get libdivsufsort separately. I notice the repo appears to have a non-empty libdivsufsort, IIUC.

Or ... should I be downloading sdsl from somewhere else? My present goal is Siren's gcsa, which needs sdsl.  I expect to be more interested in sdsl itself down the road, but is there a stable sdsl snapshot somewhere that I'd have less trouble building that would be sufficient for gcsa?

Thanks,
Bob H

$ gcc --version
gcc (Debian 4.9.2-10) 4.9.2
 ... snip ...
$
$ git --version
git version 1.7.8.2
$ git clone https://github.com/simongog/sdsl-lite.git
Cloning into 'sdsl-lite'...
remote: Counting objects: 12981, done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 12981 (delta 0), reused 0 (delta 0), pack-reused 12978
Receiving objects: 100% (12981/12981), 9.08 MiB | 10.50 MiB/s, done.
Resolving deltas: 100% (7396/7396), done.
$
$ cd sdsl-lite
$ ./install.sh /Users/bharris/sdsl/install
Library will be installed in '/Users/bharris/sdsl/install'
Copy pre-commit into .git/hooks
-- The CXX compiler identification is GNU 4.9.2
-- The C compiler identification is GNU 4.9.2
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Compiler is recent enough to support C++11.
-- Performing Test HAVE_GCC_STD=C__11__WALL__WEXTRA___DNDEBUG
-- Performing Test HAVE_GCC_STD=C__11__WALL__WEXTRA___DNDEBUG - Success
-- Performing Test HAVE_GCC_O3__FFAST_MATH__FUNROLL_LOOPS
-- Performing Test HAVE_GCC_O3__FFAST_MATH__FUNROLL_LOOPS - Success
-- Performing Test HAVE_GCC_MSSE4.2
-- Performing Test HAVE_GCC_MSSE4.2 - Success
CMake Error at external/CMakeLists.txt:14 (add_subdirectory):
  The source directory

	/Users/bharris/sdsl/sdsl-lite/external/libdivsufsort

  does not contain a CMakeLists.txt file.

CMake Error at external/CMakeLists.txt:26 (add_subdirectory):
  The source directory

	/Users/bharris/sdsl/sdsl-lite/external/googletest

  does not contain a CMakeLists.txt file.

-- Looking for C++ include cstdio
-- Looking for C++ include cstdio - found
-- Could NOT find Doxygen (missing:  DOXYGEN_EXECUTABLE) 
-- Configuring incomplete, errors occurred!
See also "/Users/bharris/sdsl/sdsl-lite/build/CMakeFiles/CMakeOutput.log".
ERROR: CMake build failed.
$
$ ls external
CMakeLists.txt	d3		googletest	libdivsufsort
$ ls external/libdivsufsort
(nothing .. the directory is empty)

**Comment by simongog** [comment]
libdivsufsort is a git submodule in the project and should be handled by git in the install.sh script.
The project is downloaded, installed and build on Travis CI after each change:
https://travis-ci.org/simongog/sdsl-lite/jobs/117667989

Maybe your git version is not recent enough to handle submodules?

**Comment by rsharris** [comment]
Thanks for the quick response.

It turns out the problem was because the machine I was running install.sh on did not have git installed.  Unfortunately the fact that it is trying to clone at that point is only reported to the user if git exists.  So where the failed install told me

	CMake Error at external/CMakeLists.txt:14 (add_subdirectory):
	The source directory
	/User/bharris/development/third_party/sdsl/sdsl-lite/external/libdivsufsort
	does not contain a CMakeLists.txt file.

my successful install tells me

	Cloning into 'external/libdivsufsort'...

and then continues on.

As to why there was no git ... the setup here is many dozens of machines all sharing the same file space.  I had originally performed the git clone on the machine I normally work on.  When it turned out the gcc on that machine was insufficient, I switched to a machine that already had a newer gcc.  Turns out that machine did not have git installed (it does now).  Surely the shell would have generated an error message when the command "git" was attempted, but something must be intercepting such messages.

Thanks for your help,
Bob H
